### ISSUE TITLE
MockServer leaks file descriptor when using ForwardChainExpectation.respond(ExpectationResponseCallback)

### ISSUE THREAD

**Issue by albans** [jamesdbloom/mockserver#512]
When using `ExpectationResponseCallback` as argument to `ForwardChainExpectation#respond`, a websocket (and a file descriptor) is opened. 
This websocket is never closed, even when the client is reseted or closed. As a consequence, a sufficiently large test suite resetting expectation between every test case and setting up new expectations using callback objects fails when exceeding the maximum number of file descriptors opened.

For example after executing:
```java
MockServerClient client = new MockServerClient("localhost", 1080)
client
    .when(request().withPath("/some/path"))
    .callback(
        new ExpectationCallback() {
            @Override
            public HttpResponse handle(HttpRequest request) {
                ...
            }
        }
    );
client.close();
```
A file descriptor  open.

**Comment by albans** [comment]
I have been looking for a way to fix this issue. One solution would be to use guava

**Comment by jamesdbloom** [comment]
closing ticket because issue was fixed in merged PR #513
